pkg/injector/webhook.go

```diff
@@ -315,6 +319,30 @@ func (wh *mutatingWebhook) mustInject(pod *corev1.Pod, namespace string) (bool,
 	return false, nil
 }
 
+// getPodOutboundPortExclusionList gets a list of ports to exclude from outbound sidecar interception.
+//
+// Outbound Ports are excluded from sidecar interception when:
+// 1. The pod is explicitly annotated with a single or comma separate list of ports
+//
+// The function returns an error when it is unable to determine whether ports need to be excluded from outbound sidecar interception.
+func (wh *mutatingWebhook) getPodOutboundPortExclusionList(pod *corev1.Pod, namespace string) ([]string, error) {
+	var ports []string
+	// Check if the pod is annotated for outbound port exclusion
+	podOutboundPortExclusionAnnotationExists, ports, err := isAnnotatedForOutboundPortExclusion(pod.Annotations, pod.Kind, fmt.Sprintf("%s/%s", pod.Namespace, pod.Name))
+	if err != nil {
+		log.Error().Err(err).Msg("Error determining outbound ports for exclusion on pod")
+		return ports, err
+	}
+
+	if podOutboundPortExclusionAnnotationExists && ports != nil {
```

Comment:
Checking for the annotation to exist is redundant here, why not just return `ports` after the initial parsing of the annotation? The returns `ports, nil` is the same in both cases.

Reply:
that's right, I and updated the code